container/go/cmd/pusher/pusher.go

```diff
@@ -63,19 +83,29 @@ func main() {
 	if *format == "docker" && filepath.Ext(*src) != ".tar" {
 		log.Fatalf("Invalid value for argument -src for -format=docker, got %q, want path to tarball file with extension .tar.", *src)
 	}
-	if *format == "legacy" && filepath.Base(*src) != "manifest.json" {
-		log.Fatalf("Invalid value for argument -src for -format=legacy, got %q, want path to manifest.json", *src)
+	if *format == "legacy" && filepath.Base(*src) != "config.json" {
+		log.Fatalf("Invalid value for argument -src for -format=legacy, got %q, want path to config.json", *src)
 	}
 	if *format == "oci" && filepath.Base(*src) != "index.json" {
 		log.Fatalf("Invalid value for argument -src for -format=oci, got %q, want path to index.json", *src)
 	}
 	if *format == "oci" || *format == "legacy" {
 		imgSrc = filepath.Dir(*src)
+		log.Printf("Determined image source path to be %q based on -format=%q, -src=%q.", imgSrc, *format, *src)
 	}
 	if *format == "docker" {
 		imgSrc = *src
 	}
-	log.Printf("Determined image source path to be %q based on -format=%q, -src=%q.", imgSrc, *format, *src)
+	if *format == "legacy" {
+		if _, err := os.Stat(filepath.Join(imgSrc, manifestPath)); err != nil {
```

Comment:
Do we ever expect a manifest for a legacy image? I would remove this check and always generate the manifest for the legacy image. If the manifest is not supposed to exist, I would instead emit an error here if os.Stat(filepath.Join(imgSrc, manifestPath)) succeeds. My concern here is that if the legacy image has an outdated manifest, the pusher will happily use that instead of generating an updated manifest.

Reply:
I will change it to override `manifest.json` (if exists) every time. The legacy image will have a `manifest` after Winnie's `create_image_config` gets merged, then I will change this to not generate anything.